### PR TITLE
chore(release): 3.1.2, automated changelogs

### LINKED ISSUE #10: Automate generating changelogs
Automated generation of changelogs would save a lot of time it would be good to integrate it with repository.
All would be based on [semantic commit messages](https://www.conventionalcommits.org/en/v1.0.0/).

Things to do:
- [ ] Write scripts generating changelogs (markdown would be the best format), old changelogs should be preserved.
- [ ] integrate script generation with github actions (on merge to master + on publishing new release)
- [ ] (if possible) create scripts generating changelogs in Fedora [link](https://docs.fedoraproject.org/en-US/packaging-guidelines/#changelogs) and Debian-based [link](https://www.debian.org/doc/debian-policy/ch-source.html#s-dpkgchangelog) format
- [ ] Write some simple guidelines for recommended naming of merge commits

### PR DESCRIPTION
**Pull request by gombosg**
## Proposed changes 

- Use automated changelog generator, see [wiki](https://github.com/AntiMicroX/antimicrox/wiki/Release-checklist) 
- Moved old static changelog to `HISTORY.md`
- Renamed changelog to `CHANGELOG.md`
- Removed version number from `README.md` - unnecessary chore maintaining it, and the releases are only 1 click away. Now only the Cmake list version number needs to be bumped.

Tested: builds, changelog appears in about window.

### CODE REVIEW

**Comment by gombosg** [comment]
We can choose a different changelog config at any time; these are just the default, it is IMO fine for starting out.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,1340 @@
...
+## [3.1.2](https://github.com/AntiMicroX/antimicrox/tree/3.1.2) (2020-09-30)
+
+[Full Changelog](https://github.com/AntiMicroX/antimicrox/compare/3.1.1...3.1.2)
+
+**Fixed bugs:**
+
+- Right click - Turbo and Clear not working properly [\#7](https://github.com/AntiMicroX/antimicrox/issues/7)
+- Wrong order for setting sets [\#6](https://github.com/AntiMicroX/antimicrox/issues/6)
+
+**Closed issues:**
```
Comment (pktiuk) [inline CHANGELOG.md:12]: I think Closed issues shouldn't be listed here

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,1340 @@
...
+- Please release a Win Installer/.exe [\#42](https://github.com/AntiMicroX/antimicrox/issues/42)
+- \[Question\] Is "antimicro" profiles is compatible with antimicrox ? [\#41](https://github.com/AntiMicroX/antimicrox/issues/41)
+- I can not install from Fedora repositories ! [\#40](https://github.com/AntiMicroX/antimicrox/issues/40)
+- Fedora resubmission [\#29](https://github.com/AntiMicroX/antimicrox/issues/29)
+- Flatpak packaging [\#9](https://github.com/AntiMicroX/antimicrox/issues/9)
+
+**Merged pull requests:**
+
+- refactor: remove Dockerfiles [\#52](https://github.com/AntiMicroX/antimicrox/pull/52) ([gombosg](https://github.com/gombosg))
+- docs: flatpak readme [\#51](https://github.com/AntiMicroX/antimicrox/pull/51) ([gombosg](https://github.com/gombosg))
```
Comment (pktiuk) [inline CHANGELOG.md:24]: I think we should also avoid putting `docs`, into changelogs, becasue things like updating readme doesn't change anything for end-user

File: CHANGELOG.md (hunk 1 above)
Comment (pktiuk) [reply CHANGELOG.md:12]: But I think they would look good in Release description.
I will investigate merging it with CI.

Comment (gombosg) [reply CHANGELOG.md:12]: Ok, I'll remove that from the config.

File: CHANGELOG.md (hunk 2 above)
Comment (gombosg) [reply CHANGELOG.md:24]: If we have a 'Merged pull requests' section then it doesn't make sense to list only _some_ of them. So either I remove this section as well (but then why are we generating it? :slightly_smiling_face: ) or keep it as is.

Comment (pktiuk) [reply CHANGELOG.md:24]: Maybe we should have just one section: `changes` and we would put there all of viable updates.

Currently, we have duplicates for fixes (Issue + PR).

Comment (pktiuk) [reply CHANGELOG.md:24]: I thought that Conventional Commits will be used for generating these changelogs, and we changelog would be generated based mainly on them.

Comment (gombosg) [reply CHANGELOG.md:24]: Ok then I'll try something like [this](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli) tomorrow

**Pull request merged**